### PR TITLE
[idrac_firmware_info] - Enhancement to support iDRAC10

### PR DESCRIPTION
**Pull request by Saksham-Nautiyal**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

File: plugins/modules/idrac_firmware_info.py [hunk 1]
```diff
@@ -104,34 +106,55 @@
...
     try:
-        with iDRACConnection(module.params) as idrac:
-            msg = idrac.update_mgr.InstalledFirmware
+        with iDRACRedfishAPI(module.params) as idrac:
+            firmware_info = get_idrac_firmware_info(idrac, module)
     except HTTPError as err:
         module.fail_json(msg=str(err), error_info=json.load(err))
     except URLError as err:
-        module.exit_json(msg=str(err), unreachable=True)
+        module.fail_json(msg=str(err), unreachable=True)
```
Comment (ABHISHEK-SINHA10) [inline plugins/modules/idrac_firmware_info.py]: Avoid fail_json and remove unused import

File: tests/unit/plugins/modules/test_idrac_firmware_info.py [hunk 2]
```diff
@@ -19,10 +19,11 @@
 from mock import MagicMock, PropertyMock
 from pytest import importorskip
 from ansible.module_utils.urls import ConnectionError, SSLValidationError
-from ansible.module_utils.six.moves.urllib.error import HTTPError, URLError
+from urllib.error import URLError, HTTPError
 from io import StringIO
 from ansible.module_utils._text import to_text
 
+
 importorskip("omsdk.sdkfile")
```
Comment (ABHISHEK-SINHA10) [inline tests/unit/plugins/modules/test_idrac_firmware_info.py]: remove this if we don't need omsdk.

File: plugins/modules/idrac_firmware_info.py (hunk 1 above)
Comment (Saksham-Nautiyal) [reply plugins/modules/idrac_firmware_info.py]: Done

File: tests/unit/plugins/modules/test_idrac_firmware_info.py (hunk 2 above)
Comment (Saksham-Nautiyal) [reply tests/unit/plugins/modules/test_idrac_firmware_info.py]: Done

File: plugins/modules/idrac_firmware_info.py [hunk 3]
```diff
@@ -104,34 +106,52 @@
...
+    try:
+        response = idrac.invoke_request(method='GET', uri=GET_IDRAC_FIRMWARE_URI_10)
+
+        if response.status_code == 200:
+            details_response = idrac.invoke_request(method='GET', uri=GET_IDRAC_FIRMWARE_DETAILS_URI_10)
+
+            if details_response and details_response.status_code == 200 and details_response.json_data:
+                filtered_data = remove_key(details_response.json_data)
+                return filtered_data
+    except Exception:
```
Comment (sachin-apa) [inline plugins/modules/idrac_firmware_info.py:128]: Can we catch the specific exception here instead of generic.

**Comment by Saksham-Nautiyal** [comment]
closed as divergent branches were created. Unable to make more commits.

**Pull request closed without merging**